File: vscode/samurai-agent/tests/agent/code_parser/fixtures/python/sample_relationships.py
# Python Relationships Test Fixture
# Tests call graph, inheritance, and type dependencies

from typing import Optional

# Call graph test
def process_user(user_id: str) -> Optional[dict]:
    user = get_user_by_id(user_id)
    if user:
        validate_user(user)
        send_notification(user)
    return user

def get_user_by_id(user_id: str) -> Optional[dict]:
    return None

def validate_user(user: dict) -> bool:
    return True

def send_notification(user: dict) -> None:
    print('Sending notification')

# Inheritance test
class Animal:
    def __init__(self, name: str):
        self.name = name
    
    def make_sound(self) -> None:
        print('Some sound')

class Dog(Animal):
    def __init__(self, name: str, breed: str):
        super().__init__(name)
        self.breed = breed
    
    def make_sound(self) -> None:
        print('Woof!')

# Multiple inheritance / Mixins
class Serializable:
    def to_json(self) -> str:
        return '{}'

class Timestamped:
    def get_timestamp(self) -> int:
        return 0

class User(Serializable, Timestamped):
    def __init__(self, name: str):
        self.name = name

File: vscode/samurai-agent/tests/agent/code_parser/fixtures/typescript/sample_relationships.ts
// TypeScript Relationships Test Fixture
// Tests call graph, inheritance, and type dependencies

import { User } from './types';

// Call graph test
export function processUser(userId: string): User | null {
    const user = getUserById(userId);
    if (user) {
        validateUser(user);
        sendNotification(user);
    }
    return user;
}

function getUserById(id: string): User | null {
    return null;
}

function validateUser(user: User): boolean {
    return true;
}

function sendNotification(user: User): void {
    console.log('Sending notification');
}

// Inheritance test
export class Animal {
    name: string;
    
    constructor(name: string) {
        this.name = name;
    }
    
    makeSound(): void {
        console.log('Some sound');
    }
}

export class Dog extends Animal {
    breed: string;
    
    constructor(name: string, breed: string) {
        super(name);
        this.breed = breed;
    }
    
    makeSound(): void {
        console.log('Woof!');
    }
}

// Interface implementation
export interface Drawable {
    draw(): void;
}

export interface Colorable {
    setColor(color: string): void;
}

export class Shape implements Drawable, Colorable {
    draw(): void {
        console.log('Drawing shape');
    }
    
    setColor(color: string): void {
        console.log(`Setting color to ${color}`);
    }
}

// Type dependencies
export type ResponseData<T> = {
    data: T;
    metadata: Metadata;
};

export type Metadata = {
    timestamp: number;
    version: string;
};

export function fetchData<T>(url: string): ResponseData<T> {
    return {} as ResponseData<T>;
}

